autopush-common/src/reliability.rs

```diff
@@ -211,25 +226,100 @@ impl PushReliability {
                 .unwrap_or_else(|| "None".to_owned()),
             new
         );
+
+        let state_key = format!("state.{id}");
+        trace!("🔍 state key: {}", &state_key);
+        // The first state is special, since we need to create the `state_key`.
+        if new == ReliabilityState::Received {
+            trace!("🔍 Creating new record");
+            // we can't perform this in a transaction because we can only increment if the set succeeds.
+            // Create the new `state.{id}` key if it does not exist, and set the expiration.
+            let options = redis::SetOptions::default()
+                .with_expiration(redis::SetExpiry::EX(expr.unwrap_or(NO_EXPIRATION)))
+                .conditional_set(redis::ExistenceCheck::NX);
+            conn.set_options::<_, _, ()>(&state_key, new.to_string(), options)
+                .await
+                .map_err(|e| {
+                    warn!("🔍⚠️ Could not create state key: {:?}", e);
+                    ApcErrorKind::GeneralError("Could not create the state key".to_owned())
+                })?;
+            // We were able to create the key, so we can increment the count.
+            conn.hincr::<_, _, _, ()>(COUNTS, new.to_string(), 1)
+                .await
+                .map_err(|e| {
+                    warn!("🔍⚠️ Could not increment received count: {:?}", e);
+                    ApcErrorKind::GeneralError("Could not increment the received count".to_owned())
+                })?;
```

Comment:
Aren't we getting 2 increments for this case, one here and one in the transaction?